### PR TITLE
[xh] Create MongoDB Sink

### PR DESCRIPTION
**Pull request by matrixstone**
# Summary
Create MongoDB Sink

# Tests
(1) Tested by creating streaming pipeline with kafka loader and mongo db exporter. 
Screenshot of success run: 
mongo_sink_pic

Data shows in mongodb container:
mongodb_data

Support store of dictionary type in mongodb:
mongo_data_with_dictionary

(2) Unit test

cc:
@wangxiaoyou1993 @tommydangerous

### CODE REVIEW

File: mage_ai/streaming/constants.py [hunk 1]
```diff
@@ -17,3 +17,4 @@ class SinkType(str, Enum):
     KAFKA = 'kafka'
     KINESIS = 'kinesis'
     OPENSEARCH = 'opensearch'
+    MONGODB = 'mongodb'
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/constants.py:20]: nit: Could you sort it in alphabetical order? (put MONGODB above OPENSEARCH)

File: mage_ai/streaming/sinks/mongodb.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass
+from mage_ai.shared.config import BaseConfig
+from mage_ai.streaming.sinks.base import BaseSink
+from typing import Dict, List
+import time
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sinks/mongodb.py:5]: nit: could you sort the import statements in alphabetical order as well?

File: requirements.txt [hunk 3]
```diff
@@ -72,6 +72,7 @@ kafka-python==2.0.2
 opensearch-py==2.0.0
 requests_aws4auth==1.1.2
 pika==1.3.1
+pymongo==4.3.3
```
Comment (wangxiaoyou1993) [inline requirements.txt:75]: Could you add pymongo to [this list](https://github.com/mage-ai/mage-ai/blob/master/setup.py#L91) and [this list](https://github.com/mage-ai/mage-ai/blob/master/setup.py#L97) as well?

File: mage_ai/streaming/sinks/mongodb.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+class MongoDbSink(BaseSink):
+    config_class = MongoDbConfig
+
+    def init_client(self):
+        self.client = MongoClient(self.config.connection_string)
+        self.database = self.client[self.config.database_name]
+        self.collection = self.database[self.config.collection_name]
+
+    def write(self, data: Dict):
+        self.collection.insert_one({"_source": data})
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sinks/mongodb.py:25]: Is `"_source"` a special key in MongoDB document or do you just put the data in "_source" for convenience? 
I would suggest checking the type of data. If it's a dict, we just insert the data as it is.

File: mage_ai/streaming/constants.py (hunk 1 above)
Comment (matrixstone) [reply mage_ai/streaming/constants.py:20]: Done

File: mage_ai/streaming/sinks/mongodb.py (hunk 4 above)
Comment (matrixstone) [reply mage_ai/streaming/sinks/mongodb.py:25]: Updated it to store dict or if data is not dict store with a key named "data"

**Comment by matrixstone** [comment]
@wangxiaoyou1993 comments addressed. Also updated a screenshot of mongo db output in the description. PTAL.

File: mage_ai/streaming/sinks/mongodb.py (hunk 4 above)
Comment (wangxiaoyou1993) [reply mage_ai/streaming/sinks/mongodb.py:25]: Could you update the `write` method as well?

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**